app/scripts/modules/core/application/modal/validation/validateApplicationName.directive.ts

```diff
@@ -33,24 +32,27 @@ class ValidateApplicationNameController implements ng.IComponentController {
   }
 }
 
+
+@DirectiveFactory('applicationNameValidator')
```

Comment:
This decoration just takes a list of services to inject.  Replaces putting:

static get $inject() { return ['applicationNameValidator']; } 


on all the directives / controllers that are used by a directive, so yeah it has to match.  It also just news up a ng.IDirective and wraps it in the function that meets  the ng.IDirectiveFactory contract.  It's not meant to shim ng2 directives into our code